### PR TITLE
ci: add gpg signing for RPM packages

### LINKED ISSUE #1384: Sign RPM packages
## Description

https://aquasecurity.github.io/trivy-repo/ describes setting up RHEL/CentOS repos with `gpgcheck=0`. For security reasons this is not a good practice. The repo cannot be browsed as it's hosted on github pages.

Can you please provide a URL to the GPG key?

Thanks!

### PR DESCRIPTION
**Pull request by afdesk**
## Description

Added inline signatures for `RPM` packages.

we use custom configuration [GoReleaser's nFPM](https://goreleaser.com/customization/nfpm/?h=rpm), so we need to set `$NFPM_ID_RPM_PASSPHRASE` and `$GPG_FILE`.

## Related issues
Closes #1384 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: goreleaser.yml [hunk 1]
```diff
@@ -251,6 +254,19 @@ docker_manifests:
...
+  args:
+    - "--batch"
+    - "-u"
+    - "{{ .Env.GPG_FINGERPRINT }}"
+    - "--output"
+    - "${signature}"
+    - "--detach-sign"
+    - "${artifact}"
+  output: true
+  stdin: '{{ .Env.GPG_PASSPHRASE }}'
```
Comment (knqyf263) [inline goreleaser.yml:269]: Do we need this signing? If I understand correctly, isn't it relevant to rpm?

**Review by knqyf263 — COMMENTED** [review]
And I want to make sure we don't break the dpkg installation.
https://aquasecurity.github.io/trivy/v0.38/getting-started/installation/#debianubuntu-official

File: docs/getting-started/installation.md [hunk 2]
```diff
@@ -15,8 +15,9 @@ In this section you will find an aggregation of the different ways to install Tr
     [trivy]
     name=Trivy repository
     baseurl=https://aquasecurity.github.io/trivy-repo/rpm/releases/$RELEASE_VERSION/\$basearch/
-    gpgcheck=0
+    gpgcheck=1
     enabled=1
+    gpgkey=https://aquasecurity.github.io/trivy-repo/deb/public.key
```
Comment (knqyf263) [inline docs/getting-started/installation.md:20]: Should we copy the key under `trivy-repo/rpm` as it is currently located under `trivy-repo/deb`.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -120,4 +120,5 @@ jobs:
       with:
         version: v1.4.1
         args: release --skip-sign --snapshot --rm-dist --skip-publish --timeout 90m
-
+      env:
+        GPG_FILE: "nofile"
```
[the remark this answers is not in the document]
Comment (afdesk) [reply .github/workflows/test.yaml:124]: > Does it work with a file that doesn't exist?

yes. the workflow uses `--skip-sign` for GoReleaser, so it doesn't try to sign the packages.
but GoReleaser needs to have an environment variable `GPG_FILE`

**Comment by afdesk** [comment]
I've created a PR https://github.com/aquasecurity/trivy-repo/pull/25

but it should be merged after the first release with signed RPMs

File: .github/workflows/test.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply .github/workflows/test.yaml:124]: Thanks for your explanation.

File: docs/getting-started/installation.md (hunk 2 above)
Comment (knqyf263) [reply docs/getting-started/installation.md]: Merged. Thanks.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**